iot/manager/manager.go

```diff
@@ -690,6 +652,166 @@ func sendCommand(w io.Writer, projectID string, region string, registry string,
+	client, err := getClient()
+	if err != nil {
+		return nil, err
+	}
+
+	parent := fmt.Sprintf("projects/%s/locations/%s/registries/%s", projectID, region, registryID)
+	response, err := client.Projects.Locations.Registries.Devices.List(parent).GatewayListOptionsGatewayType("GATEWAY").Do()
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(response.Devices) == 0 {
+		fmt.Fprintln(w, "No gateways found")
+	} else {
+		fmt.Fprintln(w, len(response.Devices), "devices:")
+		for _, gateway := range response.Devices {
+			fmt.Fprintf(w, "\t%s\n", gateway.Id)
+		}
+	}
+
+	return response.Devices, err
+}
+
+// [END iot_list_gateways]
+
+// [START iot_bind_device_to_gateway]
+
+// bindDeviceToGateway creates an association between an existing device and gateway
+func bindDeviceToGateway(w io.Writer, projectID string, region string, registryID string, gatewayID string, deviceID string) (*cloudiot.BindDeviceToGatewayResponse, error) {
+	client, err := getClient()
+
+	parent := fmt.Sprintf("projects/%s/locations/%s/registries/%s", projectID, region, registryID)
+	bindRequest := &cloudiot.BindDeviceToGatewayRequest{
+		DeviceId:  deviceID,
+		GatewayId: gatewayID,
+	}
+
+	response, err := client.Projects.Locations.Registries.BindDeviceToGateway(parent, bindRequest).Do()
+
+	if err != nil {
+		return response, err
+	}
+
+	if response.HTTPStatusCode == 200 {
+		fmt.Fprintf(w, "Bound %s to %s", deviceID, gatewayID)
+	}
+
+	return response, err
+}
+
+// [END iot_bind_device_to_gateway]
+
+// [START unbind_device_from_gateway]
+
+// unbindDeviceFromGateway unbinds a bound device from a gateway
+func unbindDeviceFromGateway(w io.Writer, projectID string, region string, registryID string, gatewayID string, deviceID string) (*cloudiot.UnbindDeviceFromGatewayResponse, error) {
+	client, err := getClient()
+
+	parent := fmt.Sprintf("projects/%s/locations/%s/registries/%s", projectID, region, registryID)
+	unbindRequest := &cloudiot.UnbindDeviceFromGatewayRequest{
+		DeviceId:  deviceID,
+		GatewayId: gatewayID,
+	}
+
+	response, err := client.Projects.Locations.Registries.UnbindDeviceFromGateway(parent, unbindRequest).Do()
+
+	if err != nil {
+		return response, err
+	}
+
+	if response.HTTPStatusCode == 200 {
+		fmt.Fprintf(w, "Unbound %s from %s", deviceID, gatewayID)
+	}
+
+	return response, err
+}
+
+// [END unbind_device_from_gateway]
+
+// [START list_devices_for_gateway]
+func listDevicesForGateway(w io.Writer, projectID string, region string, registryID, gatewayID string) ([]*cloudiot.Device, error) {
+	client, err := getClient()
+	if err != nil {
+		return nil, err
+	}
+
+	parent := fmt.Sprintf("projects/%s/locations/%s/registries/%s", projectID, region, registryID)
+	response, err := client.Projects.Locations.Registries.Devices.List(parent).GatewayListOptionsAssociationsGatewayId(gatewayID).Do()
+
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Fprintf(w, "Devices for %s:\n", gatewayID)
```

Comment:
Move this inside the else (or after returning early in the `== 0` check) for consistency?